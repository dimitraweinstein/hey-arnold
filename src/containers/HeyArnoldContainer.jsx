import React, { Component } from 'react';
import { findCharacters } from '../services/heyArnoldApi';
import CharacterList from '../components/characters/CharacterList';

export default class HeyArnoldContainer extends Component {
  state = {
    loading: true,
    characters: []
  };

  componentDidMount() {
    findCharacters().then((characters) =>
      this.setState({ characters, loading: false })
    );
  }

  render() {
    const { loading, characters } = this.state;

    if(loading) {
      return (
        <img src="https://icon-library.com/images/ajax-loading-icon/ajax-loading-icon-2.jpg"
          alt="loading spinner"
        />
      );}
    return <CharacterList characters={characters} />;
  }
}
