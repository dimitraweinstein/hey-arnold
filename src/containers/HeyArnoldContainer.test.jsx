import React from 'react';
import { render, screen } from '@testing-library/react';
import HeyArnoldContainer from '../containers/HeyArnoldContainer';
import { MemoryRouter } from 'react-router';

describe('Hey Arnold Container', () => {
  it('renders a list of characters to the page', async () => {
    render(
      <MemoryRouter>
        <HeyArnoldContainer />
      </MemoryRouter>
    );

    await screen.getByAltText('loading spinner');

    const ul = await screen.findByRole('list', { name: 'characters' });

    expect(ul).toMatchSnapshot();
  });
});

