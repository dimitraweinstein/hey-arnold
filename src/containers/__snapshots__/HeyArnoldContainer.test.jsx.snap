// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hey Arnold Container renders a list of characters to the page 1`] = `
<ul
  aria-label="characters"
>
  <li>
    <a
      href="/5da237699734fdcb7bef8f51"
    >
      <img
        alt="5da237699734fdcb7bef8f51"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
      />
      <h2>
        Arnold Shortman
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f52"
    >
      <img
        alt="5da237699734fdcb7bef8f52"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
      />
      <h2>
        Grandpa Phil
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f53"
    >
      <img
        alt="5da237699734fdcb7bef8f53"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/7/7b/DB87AC2A-C2C2-4DAB-AECF-B9F7273C746B.png/revision/latest/scale-to-width-down/306?cb=20190816101257"
      />
      <h2>
        Grandma Gertie
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f54"
    >
      <img
        alt="5da237699734fdcb7bef8f54"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/d/d9/Mitzi.png/revision/latest/scale-to-width-down/310?cb=20171212070331"
      />
      <h2>
        Mitzi
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f55"
    >
      <img
        alt="5da237699734fdcb7bef8f55"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
      />
      <h2>
        Arnie
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f56"
    >
      <img
        alt="5da237699734fdcb7bef8f56"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639"
      />
      <h2>
        Miles Shortman
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f57"
    >
      <img
        alt="5da237699734fdcb7bef8f57"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/f/fa/Stella.jpg/revision/latest?cb=20101225143956"
      />
      <h2>
        Stella Shortman
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f58"
    >
      <img
        alt="5da237699734fdcb7bef8f58"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e0/Grandpa%27s_father.png/revision/latest/scale-to-width-down/310?cb=20190228081112"
      />
      <h2>
        Grandpa's father
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f59"
    >
      <img
        alt="5da237699734fdcb7bef8f59"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
      />
      <h2>
        Grandpa's grandfather
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f5a"
    >
      <img
        alt="5da237699734fdcb7bef8f5a"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/c/c5/Oskar_Kokoshka.png/revision/latest/scale-to-width-down/310?cb=20180707032418"
      />
      <h2>
        Oskar Kokoshka
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f5b"
    >
      <img
        alt="5da237699734fdcb7bef8f5b"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/f/fe/Suzie.jpg/revision/latest/scale-to-width-down/310?cb=20110223003447"
      />
      <h2>
        Suzie Kokoshka
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f5c"
    >
      <img
        alt="5da237699734fdcb7bef8f5c"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
      />
      <h2>
        Nancy
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f5d"
    >
      <img
        alt="5da237699734fdcb7bef8f5d"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/6/67/Ernie.jpeg/revision/latest/scale-to-width-down/310?cb=20181213181427"
      />
      <h2>
        Ernie Potts
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f5e"
    >
      <img
        alt="5da237699734fdcb7bef8f5e"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/6/60/Mr._Hyunh.png/revision/latest/scale-to-width-down/310?cb=20180707031441"
      />
      <h2>
        Mr. Hyunh
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f5f"
    >
      <img
        alt="5da237699734fdcb7bef8f5f"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/8/83/Mai_Hyunh.jpeg/revision/latest/scale-to-width-down/310?cb=20190118192457"
      />
      <h2>
        Mai Hyunh
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f60"
    >
      <img
        alt="5da237699734fdcb7bef8f60"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/8/80/Mr._Smith.jpg/revision/latest?cb=20110206233402"
      />
      <h2>
        Mr. Smith
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f61"
    >
      <img
        alt="5da237699734fdcb7bef8f61"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/762C69BE-23BA-48BA-9B14-6AF3246FCD3F.jpeg/revision/latest/scale-to-width-down/304?cb=20190106211607"
      />
      <h2>
        Lana Vail
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f62"
    >
      <img
        alt="5da237699734fdcb7bef8f62"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/9/9c/Mr._Purdy_%28character%29.jpeg/revision/latest?cb=20190125214742"
      />
      <h2>
        Mr. Purdy (character)
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f63"
    >
      <img
        alt="5da237699734fdcb7bef8f63"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e5/Helga_Pataki.png/revision/latest/scale-to-width-down/266?cb=20190816021807"
      />
      <h2>
        Helga G. Pataki
      </h2>
    </a>
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f64"
    >
      <img
        alt="5da237699734fdcb7bef8f64"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/8/89/Big_Bob.jpg/revision/latest/scale-to-width-down/310?cb=20170428021615"
      />
      <h2>
        Robert "Big Bob" Pataki
      </h2>
    </a>
  </li>
</ul>
`;
